tox.ini

```diff
@@ -0,0 +1,34 @@
+[tox]
+envlist = py37, check_codestyle, check_types
+
+[testenv]
+
+# As of twisted 16.4, trial tries to import the tests as a package (previously
+# it loaded the files explicitly), which means they need to be on the
+# pythonpath. Our sdist doesn't include the 'tests' package, so normally it
+# doesn't work within the tox virtualenv.
+#
+# As a workaround, we tell tox to do install with 'pip -e', which just
+# creates a symlink to the project directory instead of unpacking the sdist.
+usedevelop=true
+
+commands = trial tests
+
+[testenv:check_codestyle]
+skip_install = True
+deps =
+  # pin deps so we don't start failing when they are updated
+  flake8==3.8.3
```

Comment:
Not sure if it matters, but we also use `flake8-comprehensions` on Synapse. Probably should be introduced as a separate PR though.